### PR TITLE
Improve cyral_repository docs

### PR DESCRIPTION
**Pull request by wcmjunior**
## Description of the change

Improve `cyral_repository` docs improving `dynamic` and `mongodb_settings` attributes.

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review

- [ ] This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] Jira issue referenced in commit message and/or PR title

### Testing

NA

### CODE REVIEW

File: docs/resources/repository.md
Comment (ricardorey10) [inline docs/resources/repository.md:151]: The indentation of the second `- If` looks different than the first one, but this might just be my browser. Please check before merging.

File: docs/resources/repository.md
Comment (ricardorey10) [inline docs/resources/repository.md:158]: I feel like here we're kinda giving them mixed instructions. Here we say:

> If `srv_record_name` is _not_ specified, then at _least one_ node must have `repo_node.host` and `repo_node.port`.

This means that I can mix static and dynamic, right? However, above, we have:

> If `server_type=sharded` and the a `srv_record_name` is provided, then all `repo_node` blocks must be declared dynamic, otherwise all of them must be static. See `dynamic`

This means that we *cannot* mix.

So, if people read the former, they might be tempted to mix stuff (which is wrong, and which will result in the API returning an error), and if they read the latter, they will do the right thing. If they read both... then they will be confused :stuck_out_tongue_winking_eye: 

Maybe we should repeat the part about not mixing static and dynamic in the former?

File: cyral/resource_cyral_repository.go
Comment (deepak-cyral) [inline cyral/resource_cyral_repository.go:424]: Is it possible (and desirable) to specify default value (since the parameter is optional)?

File: cyral/resource_cyral_repository.go
Comment (deepak-cyral) [inline cyral/resource_cyral_repository.go:415]:
```suggestion
								"addresses of the dynamic nodes. If set to `true`, `host` and `port` must " +
```

File: cyral/resource_cyral_repository.go
Comment (deepak-cyral) [inline cyral/resource_cyral_repository.go:456]:
```suggestion
								"  - If `" + RepoMongoDBSRVRecordName + "` is specified, then all `" + RepoNodesKey + "` blocks must be declared `" +
```

Comment (wcmjunior) [reply cyral/resource_cyral_repository.go:456]: @deepak-cyral , per our previous conversation I understood that all nodes must only be dynamic for `sharded` types. Do you also mean that `replicaset` types must also use `dynamic=true` if the SRV record is provided?

File: docs/resources/repository.md
Comment (wcmjunior) [reply docs/resources/repository.md:151]: You are right. It is missing a new line.

File: docs/resources/repository.md
Comment (wcmjunior) [reply docs/resources/repository.md:158]: > This means that I can mix static and dynamic, right?
Yes, this is what I understood from this internal statement: `at least one node must be static (not dynamic) UNLESS SRV record is specified in mongodb settings`, which means that we should be able to mix dynamic and static if the SRV record is not present and as long as the first node is static.

Now I am looking at [this code](https://github.com/cyralinc/crud/blob/main/repo/repo_crud.go#L125-L156), I see that if the `srv_record_name` is **not** specified, we check:
- If cluster type is `sharded`, then all nodes must be static;
- If cluster type is `standalone`,  then a single node must be declared and it needs to be static (following [this check](https://github.com/cyralinc/crud/blob/main/repo/repo_crud.go#L155) and [this one](https://github.com/cyralinc/crud/blob/22ba2cb912d6f8cee98d2970e586d558776587f6/repo/repo_crud.go#L1549)).
- As `replicaset` is not ruled out from that code, [this line](https://github.com/cyralinc/crud/blob/main/repo/repo_crud.go#L155) allows us to have mixed configurations of static+dynamic as long as the first node is static.

By contrast, if `srv_record_name` is specified, we check [here](https://github.com/cyralinc/crud/blob/22ba2cb912d6f8cee98d2970e586d558776587f6/repo/repo_crud.go#L158-L167):
- All nodes must be dynamic.


I will update the docs accordingly to those rules. LMK if I missed something.

Comment (wcmjunior) [reply docs/resources/repository.md:158]: Given the rules above, we could simplify the documentation and move it to the proper attributes as follows:

`srv_record_name`
- If provided, then all nodes must be dynamic .

`server_type`
- If `sharded` and `srv_record_name` not provided, then all nodes must be static.
- If `standalone` and `srv_record_name` not provided, then only a single static node must be declared.
- If `replicaset` and `srv_record_name` not provided, then mixed configuration of static and dynamic nodes are allowed as long as the first node is static.

File: cyral/resource_cyral_repository.go
Comment (wcmjunior) [reply cyral/resource_cyral_repository.go:424]: AFAIK this `TypeBool` is mapped to an underlying `bool` not a `*bool`, so it should not allow nulls. In this sense, I understand that the default here should be `false` then. If that is the case, then this should be the proper value for the default correct? I will double check this, though.

File: cyral/resource_cyral_repository.go
Comment (wcmjunior) [reply cyral/resource_cyral_repository.go]: Please, see the following conversation: https://github.com/cyralinc/terraform-provider-cyral/pull/361#discussion_r1110482324

File: cyral/resource_cyral_repository.go
Comment (wcmjunior) [reply cyral/resource_cyral_repository.go:424]: Having second thoughts on my previous argument and I see now it does not make sense from the user standpoint. We should indeed provide the default values in the code as they would reflect in the docs and give a better notion to our users what is the behaviour when a given argument is not provided. Anyway, I think we should address these defaults in a different PR as this would require some tests. @ricardorey10 , could you take a stab on providing these default values to the schema?

File: cyral/resource_cyral_repository.go
Comment (deepak-cyral) [inline cyral/resource_cyral_repository.go:416]:
```suggestion
								"be empty. A node with value of this field as false considered `static`.\n" +
```

File: cyral/resource_cyral_repository.go
Comment (deepak-cyral) [inline cyral/resource_cyral_repository.go:418]: This particular point probably belongs as description for `RepoNodesKey`

**Review by deepak-cyral — APPROVED** [review]
Approving so we don't hold this up - please make changes (and accept/reject suggestions) as you see fit.

File: cyral/resource_cyral_repository.go
Comment (ricardorey10) [reply cyral/resource_cyral_repository.go:424]: Sure thing. I'll put my thoughts here for discussion, and we can create a PR later (I'm a bit tied up with some Snowflake stuff now so I can't volunteer to do that change right now, but if it can wait a few days I can take it):

- "host": this should default to an empty string, as we cannot guess anything better.
- "port": this should default to zero, as we cannot guess anything.
- "dynamic": I think the default should be `false`. The reasoning here is that, if I supply the host and port but I don't supply `dynamic`, then this will result in an API error, because we can't have `dynamic = true` and non-empty host:port. This will also make it obvious that, for dynamic repo nodes, the *only* thing they have to configure is the dynamic-ness os the node (as `true`).

With the above defaults, we can have the following configurations:

```
repo_node {
  dynamic = true   // Host and port are empty, which is fine
}
```

```
repo_node {
  host = "something"    // dynamic is false, which is what we want
  port = 1234
}
```

```
repo_node {
  host = "something"    // Just combining the two above in a single config
  port = 1234
}

repo_node {
  dynamic = true
}
```

File: cyral/resource_cyral_repository.go
Comment (ricardorey10) [inline cyral/resource_cyral_repository.go:461]: This should default to an empty string. The backend will interpret that as "no SRV record supplied" and thus it will go through the "hard-coded" path.

File: cyral/resource_cyral_repository.go
Comment (ricardorey10) [inline cyral/resource_cyral_repository.go:440]: Here, the ideal behavior should be "if the server type is replicaset, then this is actually mandatory". If we can express that in code, that would be nice.

Regardless, the default here should just be an empty string.

**Review by ricardorey10 — APPROVED** [review]
@deepak-cyral @wcmjunior I left some comments about the defaults.

Regardless of those default values, though, I think this PR looks good for merging, and we can take the default stuff in a clean PR after this is merged.

**Pull request merged**
